contracts/Remittance.sol

```diff
@@ -0,0 +1,101 @@
+
+import "./SafeMath.sol";
+import "./Stoppable.sol";
+
+contract Remittance is Stoppable {
+    using SafeMath for uint;
+    
+    struct Transaction {
+        uint time;
+        address emitter;
+        uint amount;
+    }
+    
+    uint constant TX_FEES = 2000;
+    uint constant CANCELLATION_DELAY = 5 days;
+    
+    mapping(bytes32 => Transaction) private transactions;
+    uint private fees;
+    
+    event LogNewTransaction(address indexed emitter, address exchange, uint amount);
+    event LogTakeFees(address indexed emitter, uint amount);
+    event LogCancelTransaction(address indexed emitter, address exchange, uint amount);
+    event LogWithdraw(address indexed emitter, uint amount);
+    
+    function hashOTP(address exchange, bytes32 password) private view returns(bytes32 hash) {
+        require(exchange != address(0) && password != 0, "Both exchange and password must be set");
+        hash = keccak256(abi.encodePacked(address(this), exchange, password));
+    }
+    
+    function newTransaction(address exchange, bytes32 password) _onlyIfRunning public payable returns(bool success) {
+        require(exchange != address(0) && password != 0 && msg.value > 0, "An error occured. Ensure that both exchange and password are set and that your transaction value is above 0");
+        
+        bytes32 hash = hashOTP(exchange, password);
+        
+        require(transactions[hash].emitter == address(0), "Password already used");
+        
+        uint amount = msg.value;
+        if (getOwner() != msg.sender) {
+            fees = fees.add(TX_FEES);
+            amount = amount.sub(TX_FEES);
+            emit LogTakeFees(msg.sender, TX_FEES);
+        }
+        
+        emit LogNewTransaction(msg.sender, exchange, amount);
+        transactions[hash] = Transaction({time: now.add(CANCELLATION_DELAY), emitter: msg.sender, amount: amount});
+        
+        return true;
+    }
+    
+    function cancelTransaction(address exchange, bytes32 password) public returns(bool success) {
+        require(exchange != address(0) && password != 0, "Both exchange and password must be set");
+        
+        bytes32 hash = hashOTP(exchange, password);
+        Transaction memory tx = transactions[hash];
+        
+        require(tx.amount > 0, "Neither the transaction does not exist or the password is wrong");
+        require(now >= tx.time, "You must wait 5 days before cancelling the transaction");
+        require(tx.emitter == msg.sender, "The call must be initiated by the wanted exchange");
+        
+        transactions[hash].amount = 0;
+        transactions[hash].time = 0;
+        transactions[hash].emitter = address(0);
+        
+        emit LogCancelTransaction(msg.sender, exchange, tx.amount);
+        msg.sender.transfer(tx.amount);
+        
+        return true;
+    }
+    
+    function withdraw(bytes32 password) public returns(bool success) {
+        require(password != 0, "Password must be set");
```

Comment:
This check is not terribly important. After all, if it is wrong, it will fail on line 77.
So if you remove it, it will be cheaper for successful withdraws, and more expensive for mistakes.